src/components/product.js

```diff
@@ -65,7 +75,7 @@ export default class Product extends Component {
    * @param {Object} props - data and utilities passed down from UI instance.
    */
   constructor(config, props) {
-    super(config, props);
+    super(normalizeProductId(config), props);
```

Comment:
Also, we should probably be preferentially using `storefrontId` everywhere, and deprecate the use of `id`. This can be a different PR though.